From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: wu xia <xia.wu@intel.com>
Date: Wed, 10 Jun 2015 15:57:12 +0800
Subject: [PATCH] mm: export some vm_area APIs

EXPORT_SYMBOL_GPL({find_vm_area,get_vm_area_caller,
unmap_kernel_range});

These APIs are needed in DMA mapping API implementation
as a module. Device specific IOMMUs with associated DMA
mapping implementations should be buildable as modules.

Signed-off-by: Sakari Ailus <sakari.ailus@linux.intel.com>
Signed-off-by: Jianxu Zheng <jian.xu.zheng@intel.com>
Signed-off-by: wu xia <xia.wu@intel.com>
---
 mm/vmalloc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 27acb8e5aacd..c0d6de6c693c 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -1462,6 +1462,7 @@ struct vm_struct *get_vm_area_caller(unsigned long size, unsigned long flags,
 	return __get_vm_area_node(size, 1, flags, VMALLOC_START, VMALLOC_END,
 				  NUMA_NO_NODE, GFP_KERNEL, caller);
 }
+EXPORT_SYMBOL_GPL(get_vm_area_caller);
 
 /**
  *	find_vm_area  -  find a continuous kernel virtual area
@@ -1481,6 +1482,7 @@ struct vm_struct *find_vm_area(const void *addr)
 
 	return NULL;
 }
+EXPORT_SYMBOL_GPL(find_vm_area);
 
 /**
  *	remove_vm_area  -  find and remove a continuous kernel virtual area
-- 
https://clearlinux.org

